### PR TITLE
XW-2761 | check if bg image url contains latest revision

### PR DESCRIPTION
**Pull request by rybmat**
temporary fix for https://wikia-inc.atlassian.net/browse/MAIN-10545

@Wikia/x-wing 
@ludwikkazmierczak

### CODE REVIEW

File: extensions/wikia/ThemeDesigner/ThemeSettings.class.php [hunk 1]
```diff
@@ -398,7 +398,8 @@ public function getCommunityHeaderBackgroundUrl(): string {
 		$thumbnailUrl = '';
 		$originalUrl = $this->getSettings()['community-header-background-image'];
 
-		if ( VignetteRequest::isVignetteUrl( $originalUrl ) ) {
+		// @todo fix the issue with vignette urls without /latest and remove second condition
```
Comment (WikiaTech) [inline extensions/wikia/ThemeDesigner/ThemeSettings.class.php]: INFO Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 16 issues
* ![MAJOR][MAJOR] 7 major
* ![MINOR][MINOR] 7 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ThemeSettings.class.php#L173](https://github.com/Wikia/app/blob/3598f4bbff7fa24dfe54b1fc7c0b4f989baf36f7/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L173): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)
1. ![MAJOR][MAJOR] [ThemeSettings.class.php#L222](https://github.com/Wikia/app/blob/3598f4bbff7fa24dfe54b1fc7c0b4f989baf36f7/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L222): The Cyclomatic Complexity of this function "saveSettings" is 23 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [ThemeSettings.class.php#L223](https://github.com/Wikia/app/blob/3598f4bbff7fa24dfe54b1fc7c0b4f989baf36f7/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L223): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ThemeSettings.class.php#L315](https://github.com/Wikia/app/blob/3598f4bbff7fa24dfe54b1fc7c0b4f989baf36f7/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L315): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [ThemeSettings.class.php#L321](https://github.com/Wikia/app/blob/3598f4bbff7fa24dfe54b1fc7c0b4f989baf36f7/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L321): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [ThemeSettings.class.php#L327](https://github.com/Wikia/app/blob/3598f4bbff7fa24dfe54b1fc7c0b4f989baf36f7/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L327): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [ThemeSettings.class.php#L332](https://github.com/Wikia/app/blob/3598f4bbff7fa24dfe54b1fc7c0b4f989baf36f7/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L332): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MINOR][MINOR] [ThemeSettings.class.php#L5](https://github.com/Wikia/app/blob/3598f4bbff7fa24dfe54b1fc7c0b4f989baf36f7/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L5): Rename this constant "WikiFactorySettings" to match the regular expression ^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS115)
1. ![MINOR][MINOR] [ThemeSettings.class.php#L7](https://github.com/Wikia/app/blob/3598f4bbff7fa24dfe54b1fc7c0b4f989baf36f7/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L7): Rename this constant "WikiFactoryHistory" to match the regular expression ^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS115)
1. ![MINOR][MINOR] [ThemeSettings.class.php#L9](https://github.com/Wikia/app/blob/3598f4bbff7fa24dfe54b1fc7c0b4f989baf36f7/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L9): Rename this constant "HistoryItemsLimit" to match the regular expression ^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS115)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: extensions/wikia/ThemeDesigner/ThemeSettings.class.php [hunk 2]
```diff
@@ -398,12 +398,17 @@ public function getCommunityHeaderBackgroundUrl(): string {
 		$thumbnailUrl = '';
 		$originalUrl = $this->getSettings()['community-header-background-image'];
 
-		if ( VignetteRequest::isVignetteUrl( $originalUrl ) ) {
-			$thumbnailUrl = VignetteRequest::fromUrl( $originalUrl )
-				->zoomCrop()
-				->width( self::COMMUNITY_HEADER_BACKGROUND_WIDTH )
-				->height( self::COMMUNITY_HEADER_BACKGROUND_HEIGHT )
-				->url();
+		// @todo fix the issue with vignette urls without /latest and remove try-catch
```
Comment (WikiaTech) [inline extensions/wikia/ThemeDesigner/ThemeSettings.class.php]: INFO Complete the task associated to this "TODO" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1135)

File: extensions/wikia/ThemeDesigner/ThemeSettings.class.php [hunk 3]
```diff
@@ -398,12 +398,17 @@ public function getCommunityHeaderBackgroundUrl(): string {
...
+		try {
+			if ( VignetteRequest::isVignetteUrl( $originalUrl ) ) {
+				$thumbnailUrl = VignetteRequest::fromUrl( $originalUrl )
+					->zoomCrop()
+					->width( self::COMMUNITY_HEADER_BACKGROUND_WIDTH )
+					->height( self::COMMUNITY_HEADER_BACKGROUND_HEIGHT )
+					->url();
+			}
+		} catch (Exception $e) {
```
Comment (WikiaTech) [inline extensions/wikia/ThemeDesigner/ThemeSettings.class.php]: MAJOR Either remove or fill this block of code. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS108)

**Comment by WikiaTech** [comment]
SonarQube analysis reported 17 issues
* ![MAJOR][MAJOR] 8 major
* ![MINOR][MINOR] 7 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [ThemeSettings.class.php#L173](https://github.com/Wikia/app/blob/66765fa2cc181174f7dfd6465f04fc80cd2f8047/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L173): Take the required action to fix the issue indicated by this "FIXME" comment. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1134)
1. ![MAJOR][MAJOR] [ThemeSettings.class.php#L222](https://github.com/Wikia/app/blob/66765fa2cc181174f7dfd6465f04fc80cd2f8047/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L222): The Cyclomatic Complexity of this function "saveSettings" is 23 which is greater than 20 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1541)
1. ![MAJOR][MAJOR] [ThemeSettings.class.php#L223](https://github.com/Wikia/app/blob/66765fa2cc181174f7dfd6465f04fc80cd2f8047/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L223): Refactor this code to rely on RequestContext instead of context-dependent globals. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=wikia-php-rules%3AContextualWgVariableUsage)
1. ![MAJOR][MAJOR] [ThemeSettings.class.php#L315](https://github.com/Wikia/app/blob/66765fa2cc181174f7dfd6465f04fc80cd2f8047/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L315): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [ThemeSettings.class.php#L321](https://github.com/Wikia/app/blob/66765fa2cc181174f7dfd6465f04fc80cd2f8047/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L321): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [ThemeSettings.class.php#L327](https://github.com/Wikia/app/blob/66765fa2cc181174f7dfd6465f04fc80cd2f8047/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L327): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MAJOR][MAJOR] [ThemeSettings.class.php#L332](https://github.com/Wikia/app/blob/66765fa2cc181174f7dfd6465f04fc80cd2f8047/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L332): Merge this if statement with the enclosing one. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS1066)
1. ![MINOR][MINOR] [ThemeSettings.class.php#L5](https://github.com/Wikia/app/blob/66765fa2cc181174f7dfd6465f04fc80cd2f8047/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L5): Rename this constant "WikiFactorySettings" to match the regular expression ^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS115)
1. ![MINOR][MINOR] [ThemeSettings.class.php#L7](https://github.com/Wikia/app/blob/66765fa2cc181174f7dfd6465f04fc80cd2f8047/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L7): Rename this constant "WikiFactoryHistory" to match the regular expression ^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS115)
1. ![MINOR][MINOR] [ThemeSettings.class.php#L9](https://github.com/Wikia/app/blob/66765fa2cc181174f7dfd6465f04fc80cd2f8047/extensions/wikia/ThemeDesigner/ThemeSettings.class.php#L9): Rename this constant "HistoryItemsLimit" to match the regular expression ^[A-Z][A-Z0-9]*(_[A-Z0-9]+)*$. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=php%3AS115)

[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Review by weronika — APPROVED** [review]

**Pull request merged**